packages/preact/src/actions/newSearch.ts

```diff
@@ -40,16 +37,7 @@ export async function newSearch(context: ActionContext, query: SearchQuery, opti
   )
 
   try {
-    let response: SearchResult
-
-    const cachedValue = loadCachedResultIfApplicable(usePersistentCache, fullQuery)
-    if (cachedValue) {
-      response = cachedValue
-    } else {
-      const queryWithDefaults = applyQueryDefaults(pageType, fullQuery)
```

Comment:
I would keep this logic here

Reply:
Refer https://github.com/Nosto/search-js/pull/163#discussion_r2058029946

Or please give me reasons to retain it here

Reply:
Applying the defaults makes the layer cleaner
* newSearch - applying defaults
* searchWithCache - interaction with caching and underlying search function

Reply:
As per the current logic, we used query without defaults as cache key https://github.com/Nosto/search-js/blob/main/packages/preact/src/actions/newSearch.ts#L51
If we include the defaults, it would add more data to the key (facets and fields). I am not sure if that's needed.